### ISSUE TITLE
Default Browser and OpenWith overrides per URL.

### ISSUE THREAD

**Issue by BlackDex** [tasomaniac/OpenLinkWith#84]
Hello there,

First of all, thanks for this nice peace of software.
I'm not sure if this is a feature already, but i wasn't able to get it to work.

I Want to use a specific browser as my default browser, Bromite for example for all URL's.
Except, that i want to use IceRaven for some other specific sites, like youtube or whatever, because of some Add-On's that it supports.

Now, if i configure OLW first to always ask which browser, and configure the youtube link to open in IceRaven, that works great, it never asks again, but still does for other links.

If i now configure OLW to by default use Bromite, it seems to open it in both applications, which is not very useful.
Now this could be a bug, or a feature, but it would be great if we could select a default browser for all, and have some exceptions for others and not open them in multiple apps at the same time.

If you need any more information, please let me know.

**Comment by tasomaniac** [comment]
Hi, thanks for the report. I designed this app to help open links with native apps. I added the browser support later on. What you describe is supported if you want to open some links with some apps and others with a browser. But it would not work reliably if both apps are browsers.

I think this is difficult to build properly. If you have any ideas, let me know.

**Comment by BlackDex** [comment]
Well, as mentioned in my previous post, it kinda works already, except that it opens two applications, namely the one specifically configured for youtube, and the default browser configured one after the other. So, in that specific case, maybe it is an option to see if there is a specific option set, and only use that.

If that part is filtered-out, so that it won't start two apps for the same link. then it could be as easy as having the option to add custom domains to the Preferred Apps sections for example, or when someone have the link in there clipboard, make it possible to assign a specific app to it.

So basically two items:
* Do not start two apps for a link, only use the most specific.
* Add an option to assign/add a Preferred App to an URL or domain.

**Comment by shadoxxhd** [comment]
I have another request that depends on this one: to be able to add rules that modify the URL before it is passed to the browser. Eg. if url contains "something.com", then open "12ft.io/<url>" in firefox
